api/src/main/java/io/opentelemetry/OpenTelemetry.java

```diff
@@ -183,6 +198,40 @@ private OpenTelemetry() {
     return null;
   }
 
+  /**
+   * Sets up the OpenTelemetry API to use the given implementation objects.
+   *
+   * <p>This method must not be called after attempting to access an API implementation object via
+   * this class (i.e., after calling methods such as {@link #getTracerProvider()}). Basically this
+   * means that the one who is in control of the main method is the only one who may call this
+   * method. Consequently, If another method calls this, this requirement applies to that method
+   * too, recursively.
+   *
+   * <p>This method must only be called once, even if you use any {@code null} arguments. Any null
+   * arguments are defaulted using the usual mechanism, falling back to the default (no-op)
+   * implementations if no SPI can be found.
+   *
+   * @param tracerProvider The {@link TracerProvider} instance to use, or {@code null} to load using
+   *     default mechanism.
+   * @param meterProvider The {@link MeterProvider} instance to use, or {@code null} to load using
+   *     default mechanism.
+   * @param contextManager The {@link CorrelationContextManager} instance to use, or {@code null} to
+   *     load using default mechanism.
+   * @throws IllegalStateException if OpenTelemetry is already initialized.
+   */
+  public static void setupExplicitImplementations(
+      @Nullable TracerProvider tracerProvider,
+      @Nullable MeterProvider meterProvider,
+      @Nullable CorrelationContextManager contextManager) {
+    synchronized (OpenTelemetry.class) { // No double-checked locking, not performance-sensitive.
```

Comment:
I'm not worried about performance, but you should never synchronize on something that is publicly accessible like a class object. If we need to synchronized, please create an internal lock Object to synchronize on.

Reply:
I need to synchronize on the same object as the already existing `getInstance`. I've heard using `.class` like this is bad practise, but I wasn't going to introduce unrelated changes in this PR.

Reply:
can you create an issue to fix it? thanks!

Reply:
#1060